### ISSUE TITLE
parfors fails to lower

### ISSUE THREAD

**Issue by stuartarchibald** [numba/numba#2498]
This fails to lower
```python
import numpy as np
from numba import njit, prange
@njit(parallel=True)
def foo(n):
    d = np.zeros(n+1)
    a = np.zeros(n)
    for i in prange(n):
        a[i] = 1
    return a

print(foo(4))
```
yields:
```python
  File "<path>numba/numba/errors.py", line 253, in new_error_context                              
    yield                                                                                                                       
  File "<path>numba/numba/lowering.py", line 216, in lower_block                                  
    self.lower_inst(inst)                                                                                                       
  File "<path>numba/numba/lowering.py", line 360, in lower_inst                                   
    parfor.lower_parfor_parallel(self, inst)                                                                                    
  File "<path>numba/numba/npyufunc/parfor.py", line 49, in _lower_parfor_parallel                 
    assert parfor.params                                                                                                        
numba.errors.LoweringError: Failed at nopython (nopython mode backend)                                                          
```
switching ` d = np.zeros(n+1)` to ` d = np.zeros(n)` works, but `d` should be dead.

**Comment by stuartarchibald** [comment]
Ping @DrTodd13 @ehsantn @ninegua

**Comment by ninegua** [comment]
Changing [line 49 in npyufunc/parfor.py](https://github.com/numba/numba/blob/master/numba/npyufunc/parfor.py#L49) from `assert parfor.params` to:

```
assert parfor.params != None
```

would fix the problem.

**Comment by ninegua** [comment]
Fix is now put in PR #2492

**Comment by stuartarchibald** [comment]
Closing as #2492 is merged.
